File: backend/app/schemas/team.py
from typing import Optional, List
from pydantic import BaseModel, ConfigDict
import datetime


class TeamMemberBase(BaseModel):
    name: str
    role: str
    team_type: str = "Office Bearers"
    department: Optional[str] = None
    year: str
    email: Optional[str] = None
    chapter: Optional[str] = "SSIT_2026"
    quote: Optional[str] = None
    photo: Optional[str] = None
    linkedin: Optional[str] = None
    github: Optional[str] = None
    bio: Optional[str] = None
    order: Optional[int] = 10
    active: Optional[bool] = True


class TeamMemberCreate(TeamMemberBase):
    pass


class TeamMemberUpdate(BaseModel):
    name: Optional[str] = None
    role: Optional[str] = None
    team_type: Optional[str] = None
    department: Optional[str] = None
    year: Optional[str] = None
    email: Optional[str] = None
    chapter: Optional[str] = None
    quote: Optional[str] = None
    photo: Optional[str] = None
    linkedin: Optional[str] = None
    github: Optional[str] = None
    bio: Optional[str] = None
    order: Optional[int] = None
    active: Optional[bool] = None


class TeamMemberResponse(TeamMemberBase):
    id: str
    created_at: Optional[datetime.datetime] = None
    updated_at: Optional[datetime.datetime] = None

    model_config = ConfigDict(from_attributes=True)


class TeamReorderItem(BaseModel):
    id: str
    order: int


class TeamReorderRequest(BaseModel):
    items: List[TeamReorderItem]

File: src/api/team.ts
import { apiClient } from "./client"

export interface TeamMember {
  id: string
  name: string
  role: string
  team_type: "Office Bearers" | "Web Development" | "Executive" | "Events" | "Design & Media" | "Editorial" | "Other"
  teamType?: string
  department?: string
  year: string
  email?: string
  chapter?: string
  quote?: string
  photo?: string
  linkedin?: string
  github?: string
  bio?: string
  order?: number
  active?: boolean
  created_at?: string
  updated_at?: string
}

export const teamApi = {
  getAll: () => apiClient<TeamMember[]>("/api/team"),
  getById: (id: string) => apiClient<TeamMember>(`/api/team/${id}`),
  create: (data: Partial<TeamMember>) =>
    apiClient<TeamMember>("/api/team", {
      method: "POST",
      body: JSON.stringify(data),
      requireAuth: true,
    }),
  update: (id: string, data: Partial<TeamMember>) =>
    apiClient<TeamMember>(`/api/team/${id}`, {
      method: "PUT",
      body: JSON.stringify(data),
      requireAuth: true,
    }),
  delete: (id: string) =>
    apiClient<{ message: string; id: string }>(`/api/team/${id}`, {
      method: "DELETE",
      requireAuth: true,
    }),
  reorder: (items: Array<{ id: string; order: number }>) =>
    apiClient<{ message: string }>("/api/team/reorder", {
      method: "POST",
      body: JSON.stringify({ items }),
      requireAuth: true,
    }),
  uploadPhoto: async (id: string, file: File) => {
    const formData = new FormData()
    formData.append("file", file)
    return apiClient<TeamMember>(`/api/team/${id}/photo`, {
      method: "POST",
      body: formData,
      requireAuth: true,
    })
  },
}
